libs/wire-api/src/Wire/API/Routes/Named.hs

```diff
@@ -19,21 +19,38 @@
 
 module Wire.API.Routes.Named where
 
+import Control.Lens ((?~))
 import Data.Kind
 import Data.Metrics.Servant
 import Data.Proxy
+import Data.String.Conversions (cs)
+import Data.Swagger
+import Data.Typeable
 import GHC.TypeLits
 import Imports
 import Servant
 import Servant.Client
 import Servant.Client.Core (clientIn)
 import Servant.Swagger
 
-newtype Named named x = Named {unnamed :: x}
+newtype Named name x = Named {unnamed :: x}
   deriving (Functor)
 
-instance HasSwagger api => HasSwagger (Named name api) where
-  toSwagger _ = toSwagger (Proxy @api)
+-- | For 'HasSwagger' instance of 'Named'.  'KnownSymbol' isn't enough because we're using
+-- types other than string literals in some places.
+class RenderableSymbol a where
+  renderSymbol :: Text
+
+instance {-# OVERLAPPABLE #-} KnownSymbol a => RenderableSymbol a where
+  renderSymbol = cs $ symbolVal (Proxy @a)
+
+instance {-# OVERLAPPING #-} (RenderableSymbol a, Typeable b) => RenderableSymbol (a, b) where
+  renderSymbol = "(" <> renderSymbol @a <> ", " <> (cs . show . typeRep $ (Proxy @b)) <> ")"
```

Comment:
These are not sufficient to prevent this error from being raised:

```
src/Wire/API/Routes/Public/Galley.hs:54:14: error:
    • No instance for (Wire.API.Routes.Named.RenderableSymbol
                         '("put", Wire.API.Team.Feature.LegalholdConfig))
        arising from a use of ‘toSwagger’
    • In the expression: toSwagger (Proxy @ServantAPI)
      In an equation for ‘swaggerDoc’:
          swaggerDoc = toSwagger (Proxy @ServantAPI)
   |
54 | swaggerDoc = toSwagger (Proxy @ServantAPI)
```

But I don't understand why?  `LegalHold` is certainly `Typeable`, because everything is implicitly; and type-level string literals (`Symbol`) are the first instance of `KnownSymbol` that comes to mind.

Reply:
5adbe2be1b0b1168d3a7cd3de53103210da7abb9  thanks @pcapriotti !